internal/cli/server.go

```diff
@@ -111,38 +110,47 @@ func ReadyRouter() *httprouter.Router {
 	return router
 }
 
-// logRequestHandler is the logging middleware for requests
-func logRequestHandler(h http.Handler, logger logr.Logger) http.Handler {
-	fn := func(w http.ResponseWriter, r *http.Request) {
+// loggingHandler is the logging middleware for requests
+func loggingHandler(h http.Handler, logger logr.Logger) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		id := fmt.Sprintf("%d", rand.Intn(10000)) // nolint:gosec // Non-crypto use
-		log := logger.WithName(id)
+		log := logger.WithName(id).WithValues(
+			"method", r.Method,
+			"uri", r.URL.String(),
+			"user", r.Header.Get("X-Webauth-User"),
+		)
 
 		// add our logger
 		ctx := r.Context()
-		ctx = context.WithValue(ctx, tracelog.CtxLoggerKey{}, log)
+		ctx = tracelog.WithLogger(ctx, log)
 		r = r.WithContext(ctx)
 
 		// log the request first, then ...
 		logRequest(r, log)
 
 		// ... call the original http.Handler
 		h.ServeHTTP(w, r)
-	}
 
-	return http.HandlerFunc(fn)
+		if log.V(15).Enabled() {
```

Comment:
not for this PR but, we need to put names to these values. What is level 15? What are we supposed to write there? Should I put my message there, lower or higher?

Reply:
So, I made my peace with those numbers. I thought we picked [logr](https://github.com/go-logr/logr), because it doesn't have log levels.

> Verbosity-levels on info logs. This gives developers a chance to indicate arbitrary grades of importance for info logs, without assigning names with semantic meaning such as "warning", "trace", and "debug." Superficially this may feel very similar, but the primary difference is the lack of semantics. Because verbosity is a numerical value, it's safe to assume that an app running with higher verbosity means more (and less important) logs will be generated.

I'm still pro log-levels, but this is also fine. No need to overthink it I guess...

Reply:
We, as developers, randomly choose a verbosity "number". But logs are not fireworks. The user doesn't simply want more. The user is probably looking for some piece of information and I doubt they will start increasing the verbosity until they find the thing they want. Me at least, I simply increase it to "100" or similar to be sure I get what I need on the first try.

Let's discuss this outside this PR though.